### DISCUSSION TITLE
How to use normalized author metadata in Pandoc template (try #2)

### DISCUSSION THREAD

**Discussion by rcannood** [quarto-dev/quarto-cli#6135] [Q&A] [answered]
### Description

This is related to #5778 where @rgaiacs wants to use `by-author` in a markdown pandoc template. @dragonstyle notes that the authors aren't normalized when rendering to markdown and asks whether there @rgaiacs has a use case.

I've got a use case! I'm trying to write an extension for converting qmd documents to the expected format for submitting a manuscript to JOSS. This fits well into my publishing workflow, as I will typically first publish qmd manuscripts to html or gfm, and want to convert it to the JOSS markdown format prior to submission to JOSS.

Here is an [example](https://joss.readthedocs.io/en/latest/submitting.html#example-paper-and-bibliography) of a JOSS formatted document:

```markdown
---
title: 'Gala: A Python package for galactic dynamics'
tags:
  - Python
  - astronomy
  - dynamics
  - galactic dynamics
  - milky way
authors:
  - name: Adrian M. Price-Whelan
    orcid: 0000-0000-0000-0000
    equal-contrib: true
    affiliation: "1, 2" # (Multiple affiliations must be quoted)
  - name: Author Without ORCID
    equal-contrib: true # (This is how you can denote equal contributions between multiple authors)
    affiliation: 2
  - name: Author with no affiliation
    corresponding: true # (This is how to denote the corresponding author)
    affiliation: 3
affiliations:
 - name: Lyman Spitzer, Jr. Fellow, Princeton University, USA
   index: 1
 - name: Institution Name, Country
   index: 2
 - name: Independent Researcher, Country
   index: 3
---

# Summary

Foo
```

I was hoping to be able to write my template as:

```markdown
---
$if(title)$title: $title$$endif$
$if(by-author)$
author:
$for(by-author)$
  - name: $by-author.name.literal$
    $if(by-author.orcid)$orcid: $by-author.orcid$$endif$
    $if(by-author.affiliations)$affiliation: $for(by-author.affiliations)$"$by-author.affiliations.ref$", $endfor$$endif$
$endfor$
$endif$
...
---
```

However, this doesn't seem to be working.

Is there already a way I can already forcibly turn on author normalization, or is there a different way for being able to access author metadata such as orcid and affiliation?

Thanks!

**Comment by cderv** [comment]
I believe this is quite specific because your output is markdown here. 

We currently explicitly remove `authors` from metadata when output is markdown, and we do not normalize
https://github.com/quarto-dev/quarto-cli/blob/7cfadb546137c32bf684431aed2e2c0d593a5b1f/src/resources/filters/modules/authors.lua#L921-L929

This was done in 108f74fd500106c342e73ab29f7661a386c59ba5 

@dragonstyle it seems there is some use case where this can be useful as Journal like JOSS now expects a `paper.md` as submission 
https://joss.readthedocs.io/en/latest/submitting.html#example-paper-and-bibliography

**Reply by dragonstyle** [reply]
Thanks for sharing this- it does indeed make sense. I'm going to open an enhancement request to explicitly enable normalization and I'll try to get to that soon!

**Reply by cscheid** [reply]
> We currently explicitly remove `authors` from metadata when output is markdown, and we do not normalize

Sorry to drive by here. I just want to make a note that this is an instance where (I believe!) that we remove the `authors` from metadata in markdown output, because we tend to assume that markdown output is used as an "intermediate" file rather than a "full output". I've had a couple of other issues come up recently about how to resolve markdown output cleanly, and I wonder if we should spend some time figuring out what to do about that.

Specifically, I wonder if we should "bless" some specific markdown variant as the "clean output" (our users have already suggested `gfm`)

**Reply by cderv** [reply, marked as answer]
> I'm going to open an enhancement request to explicitly enable normalization

For reference, issue created at https://github.com/quarto-dev/quarto-cli/issues/6139. Thanks Charles!

**Reply by rcannood** [reply]
Thanks for looking into this! I look forward to being able to create a format for JOSS using the proposed changes :relaxed:
